Source/WebCore/animation/TimelineRange.cpp

```diff
@@ -43,11 +48,125 @@ bool SingleTimelineRange::isDefault(const CSSPrimitiveValue& value, Type type)
     return value.valueID() == CSSValueNormal || (value.isPercentage() && !value.isCalculated() && percentIsDefault(value.resolveAsPercentageNoConversionDataRequired(), type));
 }
 
+Length SingleTimelineRange::defaultValue(Type type)
+{
+    return type == Type::Start ? Length(0, LengthType::Percent) : Length(100, LengthType::Percent);
+}
+
 bool SingleTimelineRange::isOffsetValue(const CSSPrimitiveValue& value)
 {
     return value.isLength() || value.isPercentage() || value.isCalculatedPercentageWithLength();
 }
 
+TimelineRangeValue SingleTimelineRange::serialize() const
+{
+    return TimelineRangeOffset { CSSPrimitiveValue::create(valueID(name))->stringValue(), offset.isPercentOrCalculated() ? CSSNumericFactory::percent(offset.percent()) : CSSNumericFactory::px(offset.value()) };
+}
+
+static const std::optional<CSSToLengthConversionData> cssToLengthConversionData(RefPtr<Element> element)
+{
+    // FIXME: investigate container query units
```

Comment:
```suggestion
    // FIXME: Investigate container query units.
```

also maybe put this before returning the `CSSToLengthConversionData`

Reply:
Fixed.